nemo/src/program_analysis/type_inference/type_requirement.rs

```diff
@@ -280,16 +302,35 @@ pub(super) fn requirements_from_aggregates_in_rules(
                 .iter()
                 .map(|term| {
                     if let PrimitiveTerm::Variable(Variable::Universal(identifier)) = term {
-                        if identifier.name().starts_with(AGGREGATE_VARIABLE_PREFIX) {
+                        // Add static output type to aggregate output variables and constructors using them
+
+                        let aggregate_output_variables_used = if identifier.name().starts_with(AGGREGATE_VARIABLE_PREFIX) {
+                            // If the term is a aggregate variable it self, add it
+                            vec![identifier]
+                        } else if identifier.name().starts_with(CONSTRUCT_VARIABLE_PREFIX) {
```

Comment:
I'm wondering:
Should we maybe keep talking about variables instead of identifiers here and add methods to variables like:
`is_aggregate_variable()` and `is_constructor_variable()` (or similar)?
No one except the variables need to know that we distinguish them using certain prefixes and I would want to hide this as good as possible :)